packages/flutter_tools/lib/src/application_package.dart

```diff
@@ -157,11 +157,25 @@ class AndroidApk extends ApplicationPackage {
     final String packageId = manifests.first.getAttribute('package');
 
     String launchActivity;
-    for (xml.XmlElement category in document.findAllElements('category')) {
-      if (category.getAttribute('android:name') == 'android.intent.category.LAUNCHER') {
-        final xml.XmlElement activity = category.parent.parent;
-        final String enabled = activity.getAttribute('android:enabled');
-        if (enabled == null || enabled == 'true') {
+    for (xml.XmlElement activity in document.findAllElements('activity')) {
+      final String enabled = activity.getAttribute('android:enabled');
+      if (enabled != null && enabled == 'false')
+        continue;
+      for (xml.XmlElement element in activity.findElements('intent-filter')) {
```

Comment:
Please add a blank line between the if-statement and the for-loop for readability purposes.